include/swift/RemoteInspection/TypeLowering.h

```diff
@@ -239,6 +239,17 @@ class EnumTypeInfo : public TypeInfo {
     return std::count_if(Cases.begin(), Cases.end(),
                          [](const FieldInfo &Case){return Case.TR != 0;});
   }
+  unsigned getNumNonEmptyPayloadCases() const {
+    auto Cases = getCases();
+    return std::count_if(Cases.begin(), Cases.end(),
+                         [](const FieldInfo &Case){
+                           // For our purposes here, assume any case
+                           // with invalid (missing) typeinfo is non-empty
+                           return Case.TR != 0
+                             && (Case.TI.getKind() == TypeInfoKind::Invalid
+                                 || Case.TI.getSize() > 0);
+                         });
```

Comment:
I wonder if it makes sense to extract the various predicates used for classifying the payload information into standalone functions that are marked as `inline` and put into a shared header.

Reply:
Good idea, but I'm not yet entirely sure what those predicates would be.  The enum support here needs a big round of refactoring.  Slava aspires to someday merge this with the type layout machinery being used in the compiler, but I fear that's much further off.